### PR TITLE
test: additional examples for default responses

### LINKED ISSUE #413: Test Harness: Default Responses
**User Story**

As a developer I want to get back a default response when no other response codes can be returned.

**ACs**
- add test cases to test harness
- add at least the cases described in the examples section
- if any of the test fails then mark that it is an invalid result (e.g. mark as skipped) and report a Jira bug
- link that bug to this card 
- Make sure to add the actual tests to TestRails

**Additional resources**

https://stoplightio.atlassian.net/wiki/spaces/EN/pages/52887637/Prism#Prism-MockServerFeatures (please find relevant row)

**Example tests**

When I send a request to an operation
And this operation is defined
And this operation has only a default response defined
Then I should get back the default response

When I send a request to an operation
And I force the response by adding __code to the path (ie http://localhost:4010/pets/123?__code=200)
And this operation has no response with such code defined
And this operation has a default response defined
Then I should get back the default response

When I send a request to an operation
And I force the response by adding __code to the path (ie http://localhost:4010/pets/123?__code=200)
And this operation has no response with such code defined
And this operation does not have default response defined
Then I should get back a response with 500 status code
And the response's payload should describe the error

When I send a request to an operation
And this operation has no response with a matching status code
And this operation does not have default response defined
Then I should get back a response with 500 status code
And the response's payload should describe the error

When I send a request to an operation
And in the path I specify ?_code=some-code (ie ?_code=200)
And the operation has such status code response defined
Then I get back a response with a status code of 200

When I send a request to an operation
And that operation has any 2xx defined
And the request was valid
Then I get back the lowest 2xx response (e.g. if 204 and 210 are defined, return 204)

When I send a request to an operation
And in the path I specify ?_code=404
And the operation has no 404 status code response defined
And the operation has 400 status code defined, but no schema or example defined in the response
Then I get back a response with a blank body with Content-Type of text/plain

When I send a request to an operation
And in the path I specify ?_code=404
And the operation has 404 status code response defined
And in this operation there is no schema or example defined in the response
Then I get back a response with Content-Type of application/json

### PR DESCRIPTION
**Pull request by lag-of-death**
This adds failing examples for #413. The passing ones are in https://github.com/stoplightio/prism/pull/465.

Closes #413

### CODE REVIEW

**Comment by XVincentX** [comment]
@philsturgeon 

Comment on these failing tests: according to the code this is failing “on purpose”. The ability to look for the closest response is implemented only for successful responses; so you can do 2XX, and you’ll get the first one. Do you want to make the change and expand it to other responses as well, such as 4XX (which is the part of this test?)

File: packages/http/package.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
     "caseless": "^0.12.0",
     "faker": "^4.1.0",
     "fp-ts": "^1.19.3",
-    "json-schema-faker": "json-schema-faker/json-schema-faker",
+    "json-schema-faker": "XVincentX/json-schema-faker",
```
Comment (philsturgeon) [inline packages/http/package.json]: Is this intentional?

Comment (XVincentX) [reply packages/http/package.json]: Hmm, this is a leftover. Will fix.

**Comment by philsturgeon** [comment]
@lag-of-death I had a go at fixing the conflicts but there is still a yarn.lock change floating around which I don't think should be there. These tests look great, if you can fix that I'll thumbs up and we're done here! 🎉

**Comment by lag-of-death** [comment]
@phil, `yarn.lock` updated now

**Comment by lag-of-death** [comment]
BLOCKED: https://github.com/stoplightio/prism/issues/413#issuecomment-513702133

File: test-harness/specs/default_responses-AC-8.oas3.txt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      responses:
+        404:
+          description: no resource
+====server====
+mock -p 4010
+====command====
+curl -i -X GET http://localhost:4010/todos?__code=404
+====expect====
+HTTP/1.1 404 Not Found
+content-type: application/problem+json
```
Comment (XVincentX) [inline test-harness/specs/default_responses-AC-8.oas3.txt]: @philsturgeon I think this is incorrect. If there's a response with no content, we should return an empty text/plain, right?

**Pull request merged**